miden-tx/src/host/mod.rs

```diff
@@ -1,35 +1,51 @@
-use alloc::{collections::BTreeMap, string::ToString};
+use alloc::{collections::BTreeMap, string::ToString, vec::Vec};
 
-use miden_lib::transaction::{TransactionEvent, TransactionKernelError};
+use miden_lib::transaction::{
+    memory::ACCT_STORAGE_ROOT_PTR, TransactionEvent, TransactionKernelError,
+};
 use miden_objects::{
-    accounts::{AccountDelta, AccountStub},
+    accounts::{AccountDelta, AccountId, AccountStorage, AccountStub},
+    assets::Asset,
+    notes::{
+        Note, NoteAssets, NoteEnvelope, NoteId, NoteInputs, NoteMetadata, NoteRecipient,
+        NoteScript, NoteTag, NoteType,
+    },
+    transaction::OutputNote,
     Digest,
 };
 use vm_processor::{
     crypto::NodeIndex, AdviceExtractor, AdviceInjector, AdviceProvider, AdviceSource, ContextId,
-    ExecutionError, Host, HostResponse, ProcessState,
+    ExecutionError, Felt, Host, HostResponse, ProcessState,
 };
 
-mod account_delta;
-use account_delta::AccountDeltaTracker;
+mod state_tracker;
+use state_tracker::StateTracker;
 
 mod account_procs;
 use account_procs::AccountProcedureIndexMap;
 
+// CONSTANTS
+// ================================================================================================
+
+pub const STORAGE_TREE_DEPTH: Felt = Felt::new(AccountStorage::STORAGE_TREE_DEPTH as u64);
+
 // TRANSACTION HOST
 // ================================================================================================
 
 /// Transaction host is responsible for handling [Host] requests made by a transaction kernel.
-///
-/// Transaction host is composed of two components:
-/// - An advice provider which is used to provide non-deterministic inputs to the transaction
-///   runtime.
-/// - An account vault delta tracker which is used to keep track of changes made to the asset
-///   of the account the transaction is being executed against.
 pub struct TransactionHost<A> {
+    /// Advice provider which is used to provide non-deterministic inputs to the transaction
+    /// runtime.
     adv_provider: A,
-    account_delta: AccountDeltaTracker,
+
+    /// Accumulates the state changes notified via events.
+    state_tracker: StateTracker,
```

Comment:
Not a strong opinion (so fine to keep as is), but I'm not sure `stack_tracker` is a better name here, mostly because it is not clear "whose" state we are tracking. If we do keep the name, I'd at least clarify it in the comments (i.e., something like "Accumulates account state changes notified via events"). But I do think something like `account_delta` or `account_delta_tracker` would be better names.